doc/DHCPv4_Gateway/DHCPv4_gateway.md

```diff
@@ -0,0 +1,127 @@
+In case of multiple interfaces, dhcprelayagent misconfigures this giaddr to any of the interfaces. 
+This change fixes this behaviour by explicitly specifying the primary interface/gateway.
+This primary subnet is stored in db as 'subnets' of VLAN.
+
+### Requirements
+
+1. Support a new member 'Subnets' of VLAN in config_db.
+2. Support parsing and assignment of subnets from minigraph to config_db. 
+3. Support specifying subnet's first address as gateway address to command line arguments to /usr/sbin/dhcrelay as -g.
+4. isc-dhcp/dhcrelay to support '-g gateway' argument, by porting an existing patch.
+
+### Architecture Design 
+
+This HLD doesn't propose any sonic architectural changes. It follows the existing architecture of sonic - open source asc-dhcp is used for its dhcpv4 requirements.
+This HLD prposes to enhance the dhcpv4 capability by avoiding ambiguity in dhcpv4 packets wrt giaddr field.
+
+### High-Level Design 
+
+This section covers the high level design of the feature/enhancement. This section covers the following points in detail.
+		
+	- Is it a built-in SONiC feature or a SONiC Application Extension? - Built in feature.
+	- What are the modules and sub-modules that are modified for this design? - isc-dhcp.
+	- What are the repositories that would be changed? - sonic-subnet/sonic-buildimage.
+	- Module/sub-module interfaces and dependencies. 
+	- SWSS and Syncd changes in detail - N/A.
+	- DB and Schema changes (APP_DB, ASIC_DB, COUNTERS_DB, LOGLEVEL_DB, CONFIG_DB, STATE_DB) - sonic-vlan/sonic-vlan/VLAN/VLAN_LIST/Subnets.
+	- Sequence diagram if required. - N/A.
+	- Linux dependencies and interface - N/A.
+	- Warm reboot requirements/dependencies - N/A.
+	- Fastboot requirements/dependencies - N/A.
+	- Scalability and performance requirements/impact - N/A.
+	- Memory requirements - N/A.
+	- Docker dependency - N/A,
+	- Build dependency if any - N/A.
+	- Management interfaces - SNMP, CLI, RestAPI, etc.,
+	- Serviceability and Debug (logging, counters, trace etc) related design
+	- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance. - N/A.
+	- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections. - ConfigDB requirement is captured below.
+
+### SAI API 
+
+No change in SAI API.
+
+### Configuration and management 
+(This section should have sub-sections for all types of configuration and management related design. Example sub-sections for "CLI" and "Config DB" are given below. Sub-sections related to data models (YANG, REST, gNMI, etc.,) should be added as required.)
+
+#### Minigraph 
+
+<<Redis DB Structure | Json Style>>
+{
+"VLAN": {
+	"Vlan1000": {
+		"dhcp_servers": [
+			"192.0.0.1"
+		],
+		"members": [
+			"Ethernet0"
+		],
+		"vlanid": "1000",
+		“subnets”: “192.168.0.1/27”
```

Comment:
-> This change fixes this behaviour by explicitly specifying the primary interface/gateway
If your intention is to specify an IP address from the interface for the gateway address, why cant we just specify that particular IP address in the new field?
I dont understand the intention of subnets with just an IP and prefix len.

Reply:
'Subnets' of a vlan is a defined member in minigraph (eg [here](https://github.com/sonic-net/sonic-mgmt/blob/43b12cb120e00bf8beb41fad039b37de55e23e90/ansible/minigraph/switch-t0.xml#L205)). instead of adding a new field, we thought on using the same. This field can also be used in future for other purposes.